src/python/pants/backend/python/target_types_rules.py

```diff
@@ -182,6 +185,57 @@ async def generate_targets_from_python_test_utils(
     )
 
 
+class GenerateTargetsFromPexBinariesFromEntryPoints(GenerateTargetsRequest):
+    generate_from = PexBinariesFromEntryPointsGeneratorTarget
+
+
+@rule
+async def generate_targets_from_pex_binaries_from_entry_points(
+    request: GenerateTargetsFromPexBinariesFromEntryPoints,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
+    generator_addr = request.generator.address
+    entry_points_field = request.generator[PexEntryPointsField].value or []
+    overrides = request.generator[OverridesField].flatten()
+
+    # @TODO: Check for overlap:
```

Comment:
Opening issue for @TODO

Reply:
I don't think I'm that worried about overlap like this. Also, we prefer the format `file.py` over `path.to.app` because we need it for "file arguments" to work

You could add a comment saying

> Note that we don't overlap because it seems unlikely to be a problem. If it does, we should add this check.